docs/rollup-boost-ha.md

```diff
@@ -0,0 +1,328 @@
+  - [`n` Block Builders](#n-block-builders)
+    - [Failure Scenarios](#failure-scenarios-1)
+    - [Pros/Cons](#pros-and-cons-1)
+  - [Builder Load Balancer](#builder-load-balancer)
+    - [Failure Scenarios](#failure-scenarios-2)
+    - [Pros/Cons](#pros-and-cons-2)
+- [Additional Considerations](#additional-considerations)
+
+# Context/Scope
+
+The current OP Stack sequencer HA design relies on `op-conductor` to manage a cluster of sequencers. Each node runs a local conductor instance, and the cluster forms a Raft quorum to elect a single leader responsible for block production. The conductor continuously monitors the health of the sequencer, electing a new leader when the current leader is unhealthy. The leader notifies it's local sequencer to run in sequencing mode, allowing `op-node` to send FCUs with payload attributes signaling the execution client to build a new payload. All follower instances are run without sequencer mode enabled, ensuring that only one sequencer is producing blocks at a time.
+
+<p align="center">
+  <img src="../assets/op-stack-ha.png" style="width:85%;" />
+</p>
+
+With the introduction of `rollup-boost`, an additional component is introduced that sits in-between `op-node` and `op-geth` that forwards Engine API requests to an external builder.
+
+This design document outlines the architecture, components, and failure strategies required for HA `rollup-boost`. The proposed design prioritizes fault tolerance, liveliness, horizontal scalability, and minimal failover time while maintaining upstream compatibility with `op-conductor`.
+
+## Goals
+- Explore HA designs for `rollup-boost` prioritizing liveliness, fault tolerance and horizontal scalability for external block builders.
+- Maintain compatibility with `op-conductor` and its sequencing assumptions.
+
+## Non Goals
+- Define how Flashblocks are handled, consumed or streamed to the network.
+- Define how pending transactions are relayed/peered across the builders/sequencer execution clients.
+- Monitoring / alerting strategies. This can be specified in a separate document once an architecture is solidified.
+
+# Designs
+
+The following designs build on the existing HA sequencer setup by introducing a `rollup-boost` instance between each `op-node` and its local `op-geth` instance.
+
+## 1:1 Rollup Boost to Builder Deployments
+
+In this design, each `rollup-boost` instance is configured with a single external builder and default execution client. When `op-node` sends an FCU containing payload attributes, `rollup-boost` forwards the request to both the default execution client and its paired builder. Upon receiving a `get_payload` request from `op-node`, `rollup-boost` queries both the execution client and the builder. If the builder returns a payload, it is validated via a `new_payload` request sent to the default execution client. If the builder payload is invalid or unavailable, `rollup-boost` falls back to the execution client's payload.
```

Comment:
In addition to the current setup, another feature will need to be added here.

We'll need to allow conductor to proxy flashblocks web socket connections / results to the public (based on if the current node is leader or not). This is a common pattern used in conductor, proxy any interaction with the system that strictly requires to interact with the leader.

Reply:
One thing to note here is that during leadership transfer, there could be flashblocks reorg in the current design.

And if we want to prevent reorgs, we'll need to consider committing flashblocks to the raft consensus for consistency during leadership transfer, and that is second phase of our planned work